src/transport.ts

```diff
@@ -183,13 +184,31 @@ export class WebRTCDirectTransport implements Transport {
       }
     }
 
+    const eventListeningName = isFirefox ? 'iceconnectionstatechange' : 'connectionstatechange'
+
+    peerConnection.addEventListener(eventListeningName, () => {
+      switch (peerConnection.connectionState) {
+        case 'failed':
+        case 'disconnected':
+        case 'closed':
+          maConn.close().catch((err) => {
+            log.error('error closing connection', err)
+          }).finally(() => {
+            peerConnection.removeEventListener(eventListeningName, () => {})
```

Comment:
This doesn't remove this function, right? You're passing a new function here that doesn't reference any existing listener. You have to name the callback handler to remove it.

Reply:
Alternatively you can use the signal in the options to addEventListener. e.g.:

```js
const controller = new AbortController();
const signal = controller.signal;
element.addEventListener('eventname', (ev) => {
    // do stuff...
}, { signal });
// To remove the event listener:
controller.abort();
```

> Event listeners can also be removed by passing an [AbortSignal](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) to an [addEventListener()](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) and then later calling [abort()](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort) on the controller owning the signal.

https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener